public/app/features/explore/ContentOutline/ContentOutline.test.tsx

```diff
@@ -40,31 +69,68 @@ const setup = () => {
 };
 
 describe('<ContentOutline />', () => {
-  beforeEach(() => {
+  it('toggles content on button click', async () => {
     setup();
-  });
-
-  it('toggles content on button click', () => {
-    let showContentOutlineButton = screen.getByLabelText('Expand content outline');
+    let showContentOutlineButton = screen.getByRole('button', { name: 'Expand outline' });
     expect(showContentOutlineButton).toBeInTheDocument();
 
-    fireEvent.click(showContentOutlineButton);
-    const hideContentOutlineButton = screen.getByText('Collapse outline');
+    await userEvent.click(showContentOutlineButton);
+    const hideContentOutlineButton = screen.getByRole('button', { name: 'Collapse outline' });
     expect(hideContentOutlineButton).toBeInTheDocument();
 
-    fireEvent.click(hideContentOutlineButton);
-    showContentOutlineButton = screen.getByLabelText('Expand content outline');
+    await userEvent.click(hideContentOutlineButton);
+    showContentOutlineButton = screen.getByRole('button', { name: 'Expand outline' });
     expect(showContentOutlineButton).toBeInTheDocument();
   });
 
-  it('scrolls into view on content button click', () => {
-    const itemButtons = screen.getAllByLabelText(/Item/i);
+  it('scrolls into view on content button click', async () => {
+    setup();
+    const itemButtons = screen.getAllByRole('button', { name: /Item [0-9]+/ });
 
-    itemButtons.forEach((button) => {
-      fireEvent.click(button);
+    for (const button of itemButtons) {
+      await userEvent.click(button);
 
       //assert scrollIntoView is called
       expect(scrollerMock.scroll).toHaveBeenCalled();
-    });
+    }
```

Comment:
i think we missed that last time, but `toHaveBeenCalled` checks if the given function has been called at least once, i think what you want to do here is something like

```ts
    for (const button of itemButtons) {
      await userEvent.click(button);
    }
    expect(scrollerMock.scroll).toHaveBeenCalledTimes(itemButtons.length);
```

if you want to check every button results in a call to `scroll`